### PR TITLE
fix management api param processing

### PR DESCRIPTION
**Pull request by eguzki**

### CODE REVIEW

File: pkg/apis/apps/v1alpha1/apicast_types.go [hunk 1]
```diff
@@ -44,7 +44,8 @@ type APIcastSpec struct {
 	// +optional
 	CacheConfigurationSeconds *int64 `json:"cacheConfigurationSeconds,omitempty"` // APICAST_CONFIGURATION_CACHE
 	// +optional
-	ManagementAPIScope *ManagementAPIScopeType `json:"managementAPIScope,omitempty"` // MANAGEMENT_API
+	// +kubebuilder:validation:Enum=disabled;status;policies;debug
+	ManagementAPIScope *string `json:"managementAPIScope,omitempty"` // MANAGEMENT_API
```
Comment (miguelsorianod) [inline pkg/apis/apps/v1alpha1/apicast_types.go]: Nitpic. Can we fix the comment here too?

Comment (miguelsorianod) [inline pkg/apis/apps/v1alpha1/apicast_types.go]: If we change the typed string into string then the follwing content in apicast_logic_reconciler.go is not needed anymore:

```

	var managementAPIScope *string
	if r.APIcastCR.Spec.ManagementAPIScope != nil {
		res := string(*r.APIcastCR.Spec.ManagementAPIScope)
		managementAPIScope = &res
	}
```

Comment (eguzki) [reply pkg/apis/apps/v1alpha1/apicast_types.go]: :+1:

Comment (eguzki) [reply pkg/apis/apps/v1alpha1/apicast_types.go]: :+1:

**Pull request merged**